src/Artsy/Router/makeAppRoutes.tsx

```diff
@@ -44,39 +45,50 @@ export function makeAppRoutes(routeList: RouteList[]): RouteConfig[] {
     createRouteConfiguration
   )
 
+  const Component = props => {
+    const { router, setRouter } = useSystemContext()
+    // Store global reference to router instance
+    useEffect(() => {
+      if (props.router !== router) {
+        setRouter(props.router)
+      }
+
+      /**
+       * Intercept <a> tags on page and if contained within router route
+       * manifest, navigate via router versus doing a hard jump between pages.
+       */
+      catchLinks(window, href => {
+        const url = ROUTE_NAMESPACE + href
+        const foundUrl = props.router.matcher.matchRoutes(routes, url)
+        // Make sure to track pageviews when staying on the same page type.
+        // Collection pages link to each other using `RouterLink` which already
+        // handles this, so just in case avoid double triggering.
+        if (foundUrl) {
+          const toPath = url.toString()
+          // TODO: Figure out why the router prop is stale.
+          // const currentPageType = props.match.location.pathname.split("/")[1]
+          const currentPageType = window.location.pathname.split("/")[1]
+          const toPageType = toPath.split("/")[1]
+          if (currentPageType === toPageType && toPageType !== "collection") {
+            trackPageView({ path: toPath })
+          }
+          props.router.push(url)
+        } else {
+          window.location.assign(url)
+        }
+      })
+    }, [])
+
+    return <AppShell {...props} />
+  }
+
   // Return a top-level "meta" route containing all global sub-routes, which is
   // then mounted into the router.
   return [
     {
       path: ROUTE_NAMESPACE,
 
-      Component: props => {
-        const { router, setRouter } = useSystemContext()
-
-        // Store global reference to router instance
-        useEffect(() => {
-          if (props.router !== router) {
-            setRouter(props.router)
-          }
-
-          /**
-           * Intercept <a> tags on page and if contained within router route
-           * manifest, navigate via router versus doing a hard jump between pages.
-           */
-          catchLinks(window, href => {
-            const url = ROUTE_NAMESPACE + href
-            const foundUrl = props.router.matcher.matchRoutes(routes, url)
-
-            if (foundUrl) {
-              props.router.push(url)
-            } else {
-              window.location.assign(url)
-            }
-          })
-        }, [])
-
-        return <AppShell {...props} />
-      },
+      Component: withRouter(Component),
```

Comment:
This HOC prob not needed, but added anyway!

Reply:
If its not being used we should remove it